### PR TITLE
test: run sqlness test in distributed mode

### PR DESCRIPTION
**Pull request by e1ijah1**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

- run sqlness test in distributed mode
- update ci 

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: tests/runner/src/env.rs [hunk 1]
```diff
@@ -48,8 +54,16 @@ impl EnvController for Env {
...
+        let mut server = database.server_process;
+        Env::stop_server(&mut server).await;
+        if database.metasrv_process.is_some() {
+            let mut metasrv = database.metasrv_process.take().unwrap();
+            Env::stop_server(&mut metasrv).await;
+        }
+        if database.datanode_process.is_some() {
+            let mut datanode = database.datanode_process.take().unwrap();
+            Env::stop_server(&mut datanode).await;
+        }
```
Comment (MichaelScofield) [inline tests/runner/src/env.rs:66]: Can it be:

```suggestion
        if let Some(mut metasrv) = database.metasrv_process.take() {
            Env::stop_server(&mut metasrv).await;
        }
        if let Some(mut datanode) = database.datanode_process.take() {
            Env::stop_server(&mut datanode).await;
        }
```

?

**Review by MichaelScofield — COMMENTED** [review]
This is looking good to me, thx @e1ijah1 ! However, the sqlness test action is failed, can you take a look?

File: tests/runner/src/env.rs (hunk 1 above)
Comment (e1ijah1) [reply tests/runner/src/env.rs]: Nice catch, thanks

**Comment by e1ijah1** [comment]
> This is looking good to me, thx @e1ijah1 ! However, the sqlness test action is failed, can you take a look?

It looks like the datanode process failed to start. I tested it in both Linux and Mac environments but couldn't reproduce the issue. Could you please take a look at `/tmp/greptime-sqlness-datanode.log`?

**Comment by waynexia** [comment]
> @e1ijah1 Seems that log cannot be downloaded? Can you `cat` that log (and other related logs) in the sqlness post-run action so we can directly see them in the action stdout?

I'll upload the logs as CI artifacts, give me a minute

**Comment by e1ijah1** [comment]
@waynexia @MichaelScofield 

My bad. Sqlness Job doesn't start etcd, causing the issue. The problem has been resolved.

**Comment by waynexia** [comment]
There are some conflicts with #920

> The problem has been resolved.

👍

**Comment by e1ijah1** [comment]
> There are some conflicts with #920
> 
> > The problem has been resolved.
> 
> 👍

fixed, PTAL

**Review by waynexia — APPROVED** [review]
LGTM, appreciate it @e1ijah1 :heart:

**Review by MichaelScofield — APPROVED** [review]

**Pull request merged**